pkg/aws/aws_helper.go

```diff
@@ -23,17 +23,23 @@ func (c *Client) GetDNSRecord(dnsName string, hostedZoneID string) (string, erro
 		return "", fmt.Errorf("no record set exists for hostedZoneID %v with dnsName %v", hostedZoneID, dnsName)
 	}
 	recordSet := resp.ResourceRecordSets[0]
-	if len(recordSet.ResourceRecords) != 1 {
-		return "", fmt.Errorf("not exactly 1 records for hostedZoneID %v with dnsName %v", hostedZoneID, dnsName)
-	}
-	if aws.StringValue(recordSet.Name) != dnsName {
-		return "", fmt.Errorf("no record set exists for hostedZoneID %v with dnsName %v", hostedZoneID, dnsName)
-	}
-	record := aws.StringValue(recordSet.ResourceRecords[0].Value)
-	if record == "" {
-		return "", fmt.Errorf("no record set exists for hostedZoneID %v with dnsName %v", hostedZoneID, dnsName)
+	if recordSet.AliasTarget != nil {
+		// adding . to end of dnsName bc DNS adds it to record Name
+		if *recordSet.Name != dnsName+"." {
```

Comment:
the `aws` library has a function to handle pointers to strings, lets use that here. I think its `aws.StringValue(<string-pointer-here>)`. If not, this could panic.

Reply:
Also I think we were supposed check `recordSet.AliasTarget.dnsName` or something like it, not `recordSet.Name`. Can you double check that?

Reply:
sure, I'll check ```recordSet.AliasTarget.dnsName``` instead. That would be checking the record value.

![Screenshot 2023-02-09 at 11 08 31 AM](https://user-images.githubusercontent.com/19876642/217913742-0d6b8139-bfc1-4566-81e4-7eaa9dad313c.png)

<img width="785" alt="Screenshot 2023-02-09 at 11 06 39 AM" src="https://user-images.githubusercontent.com/19876642/217913845-0f3c2f19-0a06-4a9a-94e8-faf4506556e1.png">

Question for this is, do I need to check that the ```recordSet.AliasTarget.dnsName``` equals this value specifically or just that it's not empty like [here](https://github.intuit.com/kubernetes/arktika/blob/69749042f469b5b46709d51eece147c23b0f93be/pkg/controller/managedcluster/providers/aws/route53.go#L130) .